### ISSUE TITLE
Make Grouse work with snapcraft hugo

### ISSUE THREAD

**Issue by capnfabs** [capnfabs/grouse#7]
Hugo installed with snapcraft doesn't work with Grouse, because snapcraft has a super restrictive set of permissions by default that doesn't let Hugo write to / read from _existing_ temporary directories. Instead, snaps have their own private view of `/tmp` (as per [here](https://forum.snapcraft.io/t/interfaces-allow-access-tmp-directory/5129))

I don't know if I want to bother addressing this just yet -- but we ran into it in [this discussion](https://discourse.gohugo.io/t/grouse-a-diff-tool-for-hugo-sites/20835/4?u=capnfabs) and I wanted to track it.

Some ideas:

- Work out if it's possible to grant permissions / mount directories into a snap's FS layout?
- Symlink / hardlink from the temp directory to the `/home/` directory?
- Have a fallback mode that puts all grouse storage in e.g. `$HOME/grouse-diff`, which would work around this problem. But then, we'd need a strategy for cleaning everything up afterwards (temp files don't have this problem because they're deleted automatically from time-to-time)

**Comment by danyelf** [comment]
I'll add a +1 to this. With `apt-get` for Hugo still stuck at 0.68,  `snap` is the cleanest way to keep hugo up to date.

**Comment by capnfabs** [comment]
Yep! I don't have time to work on this at the moment, but (for future implementers, also if it's me!) I'd go with the last option mentioned above. The complex bit is building a strategy for cleaning up the `$HOME/grouse-diff` directory periodically, and figuring out how to integrate that with the shenanigans that grouse uses to ensure fast checkouts.
